File: AspNetMvcUnleashed/0672329980_source/Chapter08/VB/MvcApplication1/MvcApplication1/Controllers/Movie2Controller.vb
Public Class Movie2Controller
    Inherits Controller

    Private _service As IMovieService

    Sub New()
        _service = New MovieService(Me.ModelState)
    End Sub

    Sub New(ByVal service As IMovieService)
        _service = service
    End Sub

    Function Index() As ActionResult
        Return View(_service.ListMovies())
    End Function

    ' GET: /Movie/Create

    Function Create() As ActionResult
        Return View()
    End Function


    ' POST: /Movie/Create

    <AcceptVerbs(HttpVerbs.Post)> _
 Function Create(<Bind(Exclude:="Id")> ByVal movieToCreate As Movie) As ActionResult
        If _service.CreateMovie(movieToCreate) Then
            Return RedirectToAction("Index")
        End If
        Return View()
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter08/CS/MvcApplication1/MvcApplication1/Controllers/Movie2Controller.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class Movie2Controller : Controller
    {
        private IMovieService _service;


        public Movie2Controller() 
        {
            _service = new MovieService(this.ModelState);
        }

        public Movie2Controller(IMovieService service)
        {
            _service = service;
        }

        public ActionResult Index()
        {
            return View(_service.ListMovies());
        }


        //
        // GET: /Movie/Create

        public ActionResult Create()
        {
            return View();
        }

        //
        // POST: /Movie/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create([Bind(Exclude = "Id")]Movie movieToCreate)
        {
            if (_service.CreateMovie(movieToCreate))
                return RedirectToAction("Index");
            return View();
        }


    }
}
